### PR TITLE
Fix#27796 Added condition and test case for the getData parameter input functionality

### LINKED ISSUE #27796: [Good First Issue]: [OV JS] Add check for arguments to Tensor.getData()
### Context

OpenVINO works in Node.js environment!
We are looking for new contributors who can help with enabling C++ API methods 
in JavaScript side.

First of all read Node.js API [Contribution Guide](https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/js/node/CONTRIBUTING.md).

#### Task Details

The `Tensor.getData()` method, which takes no arguments, should return an error if an argument is passed.

### What needs to be done?

- modify `Tensor.getData()` method [node/src/tensor.cpp#L69](https://github.com/openvinotoolkit/openvino/blob/f2d2c221ef34f8300a1a7460c9532247d343bb54/src/bindings/js/node/src/tensor.cpp#L69)
- you can take a look at [this method](https://github.com/openvinotoolkit/openvino/blob/f2d2c221ef34f8300a1a7460c9532247d343bb54/src/bindings/js/node/src/tensor.cpp#L169)
- add a test to [this suite](https://github.com/openvinotoolkit/openvino/blob/f2d2c221ef34f8300a1a7460c9532247d343bb54/src/bindings/js/node/tests/unit/tensor.test.js#L85)

### How to take this issue

To take this issue leave text: `.take` as the comment in this issue.
It will assign this issue to you automatically. Please, make sure that the issue has the status **Contributors Needed** and doesn't have another user in Assingnees.

### Example Pull Requests

 - https://github.com/openvinotoolkit/openvino/pull/23743
 - https://github.com/openvinotoolkit/openvino/pull/24023

### Resources

- [[NEW] Contribution Guide](https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/js/node/CONTRIBUTING.md)
- [JavaScript API Developer Documentation](https://github.com/openvinotoolkit/openvino/blob/master/src/bindings/js/docs/README.md#openvino-node-package-developer-documentation)
- [Node.js API Reference](https://docs.openvino.ai/2024/api/nodejs_api/addon.html)
- [Node.js bindings examples of usage](https://github.com/nodejs/node-addon-examples)
- [What is OpenVINO?](https://github.com/openvinotoolkit/openvino#what-is-openvino-toolkit)
- [How to contributing to OpenVINO](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [User documentation](https://docs.openvino.ai/)

### Contact points

@almilosz @vishniakov-nikolai 

Please, mention us in this issue discussion if you have any questions.

### Ticket

-

### PR DESCRIPTION
**Pull request by Manideep-Kanna**
This fixes #27796 
Implemented the exception for the passing parameter scenario for getData function 

- Added condition to check if parameters are passed or not
- Added tests in Tensor.test.js

### CODE REVIEW

File: src/bindings/js/node/src/tensor.cpp [hunk 1]
```diff
@@ -66,6 +66,12 @@ Napi::Object TensorWrap::wrap(Napi::Env env, ov::Tensor tensor) {
 }
 
 Napi::Value TensorWrap::get_data(const Napi::CallbackInfo& info) {
+    Napi::Env env = info.Env();
+    if (info.Length() > 0) {
+        reportError(env, "getData() does not accept any arguments.");
+        return env.Undefind
```
Comment (almilosz) [inline src/bindings/js/node/src/tensor.cpp:72]:
```suggestion
        return env.Undefined();
```

**Review by almilosz — COMMENTED** [review]
Hi! Thanks for taking up this task! Please look at the comment :)

File: src/bindings/js/node/src/tensor.cpp (hunk 1 above)
Comment (Manideep-Kanna) [reply src/bindings/js/node/src/tensor.cpp]: Hi @almilosz, thank you for pointing out the issue. Sorry, I forgot to push the commit. Doing the push right now. Can you please review it.

File: src/bindings/js/node/src/tensor.cpp [hunk 2]
```diff
@@ -66,6 +66,12 @@ Napi::Object TensorWrap::wrap(Napi::Env env, ov::Tensor tensor) {
 }
 
 Napi::Value TensorWrap::get_data(const Napi::CallbackInfo& info) {
+    Napi::Env env = info.Env();
+    if (info.Length() > 0) {
+        reportError(env, "getData() does not accept any arguments.");
+        return env.Undefined()
```
Comment (almilosz) [inline src/bindings/js/node/src/tensor.cpp:72]:
```suggestion
        return env.Undefined();
```
there is a semicolon missing ;) I recommend checking `git status` carefully, building the project and running tests before pushing changes

Comment (Manideep-Kanna) [reply src/bindings/js/node/src/tensor.cpp]: Done @almilosz

**Review by almilosz — APPROVED** [review]

**Comment by almilosz** [comment]
build_jenkins

**Pull request merged**